File: src/dsa/Kadane.java
package dsa;

public class Kadane {

    public static int maxSubsetAdjacentSum(int[] array) {
        int maxEndingHere = array[0];
        int maxSoFar = array[0];
        for (int i = 1; i < array.length; i++) {
            int num = array[i];
            maxEndingHere = Math.max(num, maxEndingHere + num);
            maxSoFar = Math.max(maxSoFar, maxEndingHere);
        }
        return maxSoFar;
    }
}

File: src/dsa-py/Kadane.py
def max_subset_adjacent_sum(arr):
    max_ending_here = arr[0]
    max_so_far = arr[0]
    for i in range(len(arr)):
        num = arr[i]
        max_ending_here = max(num, max_ending_here + num)
        max_so_far = max(max_so_far, max_ending_here)

    return max_so_far
